### PR TITLE
drop python 2 support

### PR DESCRIPTION
**Pull request by Priyankasaggu11929**
#### What type of PR is this?
/kind feature
/kind clean-up

#### What this PR does / why we need it:
The PR makes changes to drop support for Python 2 from the next release (v18.0.0). The changes includes:
- [ ] drop Python 2 from master branch
- [ ] make sure all the tests use Python 3
- [ ] drop Python 2 from v18 beta release.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-client/python/issues/1413

#### Does this PR introduce a user-facing change?

```release-note
The `python2` support will be removed in 18.0.0 beta release. All the tests will use `python3` versions.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes-client/python/blob/master/CHANGELOG.md#v1700-snapshot
```
Signed-off-by: Priyanka Saggu <priyankasaggu11929@gmail.com>

### CODE REVIEW

**Comment by Priyankasaggu11929** [comment]
@yliaog @roycaihw 

Need advice here: https://travis-ci.org/github/kubernetes-client/python/jobs/771168725

I'm trying to update the `scripts/update-pycodestyle.sh` test file to make it compatible with python3 versions. To begin testing, I'm starting with python version 3.9

From the `.travis.yaml` file
```
  - stage: test
    python: 3.9
    env: TOXENV=update-pycodestyle
```
The `pycodestyle` when ran with python v3.9 is throwing errors & suggesting changes in the python scripts (for ex - in case of above failing job, it is prompting for changes in `examples/multiple_clusters.py` python file)

- `Q 1`: So, I want to confirm that the right way to update the `scripts/update-pycodestyle.sh` (& other tests) would be step by step correcting the example files until it goes all green for python v3.9? right?

- `Q 2`: I would have to add the `update-pycodestyle` check for all the current set of python3 versions ( py3{5,6,7,8,9} )?

- `Q 3`: What's the difference between `py3{5,6,7,8,9} ` & `py3{5,6,7,8,9}-functional`?

Thank you!

**Comment by Priyankasaggu11929** [comment]
@roycaihw , @yliaog 

So far, I have done the following to remove Python 2 support:
- removed python2 from the `.travis.yaml` file
- removed python2 from the `tox.ini` file
- removed `-y` flag from `isort` command in `update-pycodestle.sh` script (as it was throwing error for unknown flag `-y` while executing it with python3 versions)
- fixed sequence of module imports  in the following files, to fix the failing `pycodestyle` checks:
    - `examples/multiple_clusters.py`
    - `examples/pick_kube_config_context.py`
    - `examples/pod_config_list.py`
- tested `coverage` & `codecov` tests with python3 version (currently `v3.9`)
- tested updated `update-pycodestyle.sh` for python3 version `v3.9` and architecture type, `ppc6le`

With these much changes, the CI jobs are green so far.

File: .travis.yml [hunk 1]
```diff
@@ -68,18 +64,12 @@ jobs:
       distributions: sdist bdist_wheel
 
   - stage: test
-    python: 2.7
+    python: 3.9
     env: TOXENV=update-pycodestyle
-    arch: ppc64le
+    arch: ppc64le
```
Comment (yliaog) [inline .travis.yml:69]: looks there are extra spaces at the end, please remove them

File: scripts/update-pycodestyle.sh [hunk 2]
```diff
@@ -67,7 +67,7 @@ done
 
 echo "--- applying isort"
 for SOURCE in $SOURCES; do
-    isort -y $SOURCE
+    isort $SOURCE
```
Comment (yliaog) [inline scripts/update-pycodestyle.sh:70]: it should have been called before with python3, not sure why there was no errors before

**Review by yliaog — COMMENTED** [review]
please remove '[WIP]'

File: .travis.yml (hunk 1 above)
Comment (Priyankasaggu11929) [reply .travis.yml:69]: removed the spaces. Thanks!

File: scripts/update-pycodestyle.sh (hunk 2 above)
Comment (Priyankasaggu11929) [reply scripts/update-pycodestyle.sh:70]: Just to confirm, there is no action required here, right?

Thank you!

Comment (yliaog) [reply scripts/update-pycodestyle.sh:70]: no action, just wondering if you know why there was no error before

Comment (Priyankasaggu11929) [reply scripts/update-pycodestyle.sh:70]: There were no previous test running `update-pycodestyle.sh` script with python3.x versions.

File: .travis.yml (hunk 1 above)
Comment (Priyankasaggu11929) [reply .travis.yml:69]: Also, re-iterating over questions asked in this comment https://github.com/kubernetes-client/python/pull/1468#issuecomment-841440671 :

Q 2: Do I require to add the `update-pycodestyle.sh` check for all the current set of python3 versions ( py3{5,6,7,8,9} )?

Q 3: What's the difference between py3{5,6,7,8,9} & py3{5,6,7,8,9}-functional?

**Comment by Priyankasaggu11929** [comment]
Also, re-iterating over questions asked in this comment https://github.com/kubernetes-client/python/pull/1468#issuecomment-841440671 :

Q 2: Do I require to add the `update-pycodestyle.sh` check for all the current set of python3 versions ( py3{5,6,7,8,9} )?

Q 3: What's the difference between py3{5,6,7,8,9} & py3{5,6,7,8,9}-functional?

File: scripts/update-pycodestyle.sh (hunk 2 above)
Comment (yliaog) [reply scripts/update-pycodestyle.sh:70]: ok, that explains it.

**Comment by yliaog** [comment]
sorry, missed your prior questions.
Q2: you only need to run one python3 version, say python 37, as all python3 versions share one common style guide

Q3: functional has e2e tests, the commands to show the difference is here:
https://github.com/kubernetes-client/python/blob/821618feb3456674ea586047b972b3ae7559fa42/tox.ini#L15

**Comment by Priyankasaggu11929** [comment]
Thank you @yliaog. That was really helpful clarification. :)

**Pull request merged**